### PR TITLE
Record / Update only the major go version

### PR DESCRIPTION
**Pull request by freeformz**
The idea is that we shouldn't care about the minor revision of Go.

/cc @kr, @neurogeek

### CODE REVIEW

File: update.go [hunk 1]
```diff
@@ -12,28 +13,64 @@ import (
...
+)
 
 func init() {
 	cmdUpdate.Flag.BoolVar(&saveT, "t", false, "save test files during update")
+	cmdUpdate.Flag.BoolVar(&updateGoVer, "goversion", false, "update the recorded go version")
 }
 
 func runUpdate(cmd *Command, args []string) {
-	err := update(args)
+	if updateGoVer {
```
Comment (kr) [inline update.go]: Nice.

File: dep.go [hunk 2]
```diff
@@ -64,8 +64,17 @@ func uniq(a []string) []string {
...
 
-// goVersion returns the version string of the Go compiler
-// currently installed, e.g. "go1.1rc3".
+// trimGoVersion and return the major version
+func trimGoVersion(version string) (string, error) {
+	p := strings.Split(version, ".")
+	if len(p) < 2 {
+		return "", fmt.Errorf("Error determing major go version from: %q", version)
+	}
+	return strings.Join([]string{p[0], p[1]}, "."), nil
```
Comment (kr) [inline dep.go]: Why not `p[0] + "." + p[1]`?

**Comment by kr** [comment]
LGTM

**Comment by neurogeek** [comment]
LGTM too.

I would like to know what prompted this? Just clean up or is there a more 'profound' reason?
Thanks!

File: dep.go (hunk 2 above)
Comment (freeformz) [inline dep.go]: Heh. No reason. That's better, I'll change it.

**Comment by freeformz** [comment]
@neurogeek After some issue discussion (heroku/heroku-buildpack-go/100 and another I can't find atm) I generally think that we should really only care about the major version of Go and let downstream systems care about selecting the minor version (the user or build systems). There are probably exceptions to that, hence I also stopped auto-updating the go version on save, allowing people to specify whatever they want if they manually want to manage the file.

**Comment by neurogeek** [comment]
Makes sense. 
Thanks for the details.

**Comment by flavorjones** [comment]
Awesome, thanks for this!

**Pull request merged**